packages/expo-cli/src/commands/init.ts

```diff
@@ -357,24 +363,47 @@ async function installNodeDependenciesAsync(projectRoot: string, packageManager:
   }
 }
 
-export async function initGitRepoAsync(
+async function checkSkipInitializeGitTree(
```

Comment:
i would split this function out into two pieces and then move it into the action function

```js
let shouldSkipInitializeGitTree = false;
try {
  if (await isInsideGitRepo(cdPath)) {
    // code log "your project will be initialized within a ... blah blah"
    shouldSkipInitializeGitTree = // code to prompt for skipping
  }
} catch(e) {
  if (e.code === 'GIT_NOT_INSTALLED') { // throw an error with this code from isInsideGitRepo
    // `git` not found in PATH, skipping initializing a git repository for project.
    shouldSkipInitializeGitTree = true;
  } else {
    // unexpected exception, rethrow
    throw e;
  }
}
```

Reply:
the problem with the `checkSkipInitializeGitTree` is it is doing too much. we really just want a function that checks if we're in git repo or not, and then we can prompt from action function. it's often cleaner to have prompts in the action than chase them around between different helper methods

Reply:
@brentvatne thanks for the suggestions. Working on the changes right now. That being said, I'm running into a little snag and wondering if you have any recommendations. 

So it looks like we're importing @types/node for some of our TS definitions, and specifically `Error`.

However, that definition of `Error` does not include the `code` property: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53034

So I'm mulling over alternatives:

- Create a custom Error and throw that?
- Find a way around the TS definition?
- Other solutions?

Curious if you have any thoughts or recommendations? Thanks!

Reply:
we define our own error classes in a couple places here:

- https://github.com/expo/expo-cli/blob/522b989a1c5bf6cb88ecdd7330dd50ae4fc99857/packages/expo-cli/src/CommandError.ts
- https://github.com/expo/expo-cli/blob/522b989a1c5bf6cb88ecdd7330dd50ae4fc99857/packages/json-file/src/JsonFileError.ts

we could do the same sort of thing for this error

Reply:
Thanks @brentvatne - much appreciated 🙏. I'll take a look.